daprdocs/content/en/operations/components/setup-pubsub/supported-pubsub/setup-gcp-pubsub.md

```diff
@@ -52,19 +54,22 @@ The above example uses secrets as plain strings. It is recommended to use a secr
 
 | Field              | Required | Details | Example |
 |--------------------|:--------:|---------|---------|
-| type           | Y | GCP credentials type  | `service_account`
-| project_id     | Y | GCP project id| `projectId`
-| private_key_id | Y | GCP private key id | `"privateKeyId"`
-| private_key    | Y | GCP credentials private key. Replace with x509 cert | `12345-12345`
-| client_email   | Y | GCP client email  | `"client@email.com"`
-| client_id      | Y |  GCP client id | `0123456789-0123456789`
-| auth_uri       | Y | Google account OAuth endpoint | `https://accounts.google.com/o/oauth2/auth`
-| token_uri      | Y | Google account token uri | `https://oauth2.googleapis.com/token`
-| auth_provider_x509_cert_url | Y | GCP credentials cert url | `https://www.googleapis.com/oauth2/v1/certs`
-| client_x509_cert_url | Y | GCP credentials project x509 cert url | `https://www.googleapis.com/robot/v1/metadata/x509/<PROJECT_NAME>.iam.gserviceaccount.com`
+| type           | N | GCP credentials type. Only `service_account` is supported. Defaults to `service_account`  | `service_account`
+| projectId     | Y | GCP project id| `myproject-123`
+| identityProjectId | N | If the GCP pubsub project is different from the identity project, specify the identity project using this attribute  | `"myproject-123"`
+| privateKeyId | N | If using explicit credentials, this field should contain the `private_key_id` field from the service account json document | `"my-proviate-key"`
+
+| privateKey    | N |  If using explicit credentials, this field should contain the `private_key` field from the service account json | `-----BEGIN PRIVATE KEY-----`
```

Comment:
I'd paste an example of what a private key would look like here

Reply:
yeah. This is the start of one, a private key is super-long so it would mess up the table flow if pasted in its entirety. But I can add a little bit more